ksy_schema.json

```diff
@@ -352,6 +352,10 @@
             }
           ]
         },
+        "valid": {
+          "description": "expected value that the parser should encounter at this point (or throw an error). Unlike `contents`, this is the value *after* parsing.",
```

Comment:
> This entire sentence is weird and inaccurate (I don't like "expected value", "this point" or "(or throw an error)"). Instead of explaining how, I'll rather come up with something myself when I have time.

Agreed. I don't understand this very well so I just took a quick stab with my limited understanding. You're right - `valid` is considerably more flexible, and I'll update to fix it.

> Unlike contents, this is the value after parsing.
>> What do you mean by this?

For instance, I have a value that's expected to be `0x0000ffff`. `{ type: u4, valid: 0x0000ffff }` checks that value after parsing (perhaps a better word would be "casting") it to its declared type (i.e. `u4').

---

I think the biggest limit to my understanding with the schema is that it's not quite clear which strings are supposed to be expressions, identifiers, or literal strings.